library/sha3.c

```diff
@@ -14,6 +14,30 @@
 
 #if defined(MBEDTLS_SHA3_C)
 
+/*
+ * These macros select manually unrolled implementations of parts of the main permutation function.
+ *
+ * Unrolling has a major impact on both performance and code size. gcc performance benefits a lot
+ * from manually unrolling at higher optimisation levels.
+ *
+ * Rolling up the theta loop saves a lot of code-size at small performance cost. The code-size
+ * saving then enables us to unroll the other loops for a net code-size saving with a net
+ * performance win.
+ *
+ * Depending on your compiler and target, it may be beneficial to adjust these; the defaults here
+ * should give sensible trade-offs for gcc and clang.
+ */
+#undef   MBEDTLS_SHA3_THETA_UNROLL //no-check-names
```

Comment:
You can avoid the need for `//no-check-names` by giving the `.c`-file-internal macros a name that doesn't have with `MBEDTLS_`. That's just for information, I have no particular preference one way or the other.